### PR TITLE
[TASK] Check mod.web_list.allowedNewTables/deniedNewTables in admin m…

### PR DESCRIPTION
**Pull request by Mabahe**
…odule

With Page TSconfig
mod.web_list {
   allowedNewTables = foo
   deniedNewTables =
}
you can control which record types get a new link in the new record view.
With this patch the news administration module checks this settings for
the "new" buttons in document header, too.

Resolves: #247

### CODE REVIEW

**Comment by georgringer** [comment]
Nice, Thx! Can you set the third parameter for in_array to true?

File: Classes/Controller/AdministrationController.php [hunk 1]
```diff
@@ -107,6 +107,19 @@ protected function initializeView(ViewInterface $view)
         $pageRenderer->loadRequireJsModule('TYPO3/CMS/News/AdministrationModule');
         $dateFormat = ($GLOBALS['TYPO3_CONF_VARS']['SYS']['USdateFormat'] ? ['MM-DD-YYYY', 'HH:mm MM-DD-YYYY'] : ['DD-MM-YYYY', 'HH:mm DD-MM-YYYY']);
         $pageRenderer->addInlineSetting('DateTimePicker', 'DateFormat', $dateFormat);
+
+        $web_list_modTSconfig = BackendUtilityCore::getModTSconfig($this->pageUid, 'mod.web_list');
+        $this->allowedNewTables = GeneralUtility::trimExplode(
```
Comment (georgringer) [inline Classes/Controller/AdministrationController.php]: those properties are undefined.

File: Classes/Controller/AdministrationController.php [hunk 2]
```diff
@@ -227,6 +238,24 @@ protected function createButtons()
...
+        if (!$this->getBackendUser()->check('tables_modify', $table)) {
+            return false;
+        }
+
+        // No deny/allow tables are set:
+        if (empty($this->allowedNewTables) && empty($this->deniedNewTables)) {
+            return true;
+        }
+
+        return !in_array($table, $this->deniedNewTables) && (empty($this->allowedNewTables) || in_array($table, $this->allowedNewTables));
```
Comment (georgringer) [inline Classes/Controller/AdministrationController.php]: please add true as 3rd parameter to in_array calls

**Comment by Mabahe** [comment]
Thx, will fix this in the next days

**Comment by georgringer** [comment]
do you plan to keep working on it?

File: Classes/Controller/AdministrationController.php (hunk 1 above)
Comment (Mabahe) [inline Classes/Controller/AdministrationController.php]: right, defined now

File: Classes/Controller/AdministrationController.php (hunk 2 above)
Comment (Mabahe) [inline Classes/Controller/AdministrationController.php]: Added

**Comment by Mabahe** [comment]
right, thx for the ping

**Pull request merged**